packages/nextjs/src/server/utils/getAuth.ts

```diff
@@ -0,0 +1,50 @@
+import { createGetToken } from '@clerk/backend-core';
+import { AuthStatus } from '@clerk/backend-core';
+import { NextRequest } from 'next/server';
+
+import { AuthStatusMapping, SessionsApi } from '../types';
+
+export function createGetAuth(sessionsApi: SessionsApi) {
+  return function (req: NextRequest) {
+    // When the __clerk query param is set, we trust that middleware already ran
+    // Then, we don't have to run any re-verify the JWT here, we
+    // can just strip out the claims manually.
+
+    const clerkParam = req.nextUrl.searchParams.get('__clerk');
+
+    if (!clerkParam) {
+      throw 'You need to use "runClerkMiddleware" on your Next.js middleware file.';
+    }
+
+    // Signed out case
+    if (clerkParam === AuthStatusMapping[AuthStatus.SignedOut]) {
+      return {
+        // TODO what to return here?
+        sessionClaims: null,
```

Comment:
Addressing the TODO here, do we simply want to return the signed out state as in: https://github.com/clerkinc/javascript/blob/0889bde9bc7f9e1a5d4c1e706c49212e1f7b36f4/packages/backend-core/src/util/createGetToken.ts#L41

or am I missing something?

Reply:
You're probably right